src/commandLine.js

```diff
@@ -0,0 +1,243 @@
+        if (!isNaN(probablePort) && (probablePort <= constants.PORTS.MAX) && (probablePort >= constants.PORTS.MIN)) {
+          RdGlobalConfig.RD_HANDLER_PORT = probablePort;
+        } else {
+          console.log("\nPort can only range from:", constants.PORTS.MIN, "to:", constants.PORTS.MAX);
+          invalidArgs.add('--port');
+        }
+        argv.splice(index, 2);
+      } else {
+        invalidArgs.add('--port');
+        argv.splice(index, 1);
+      }
+    }
+
+    // delay for retries in case of request failures
+    index = argv.indexOf('--retry-delay');
+    if (index !== -1) {
+      if (CommandLineManager.validArgValue(argv[index + 1])) {
+        var probableDelay = parseFloat(argv[index + 1]);
+        if (!isNaN(probableDelay) && probableDelay >= 0) {
+          RdGlobalConfig.RETRY_DELAY = probableDelay;
+        } else {
+          console.log("\nDelay for retries should be a valid number");
+          invalidArgs.add('--retry-delay');
+        }
+        argv.splice(index, 2);
+      } else {
+        invalidArgs.add('--retry-delay');
+        argv.splice(index, 1);
+      }
+    }
+
+    // timeout for requests being fired from the tool
+    index = argv.indexOf('--request-timeout');
+    if (index !== -1) {
+      if (CommandLineManager.validArgValue(argv[index + 1])) {
+        var clientRequestDelay = parseFloat(argv[index + 1]);
+        if (!isNaN(clientRequestDelay) && clientRequestDelay >= 0) {
+          RdGlobalConfig.CLIENT_REQ_TIMEOUT = clientRequestDelay;
+        } else {
+          console.log("\nTimeout for requests should be a valid number");
+          invalidArgs.add('--request-timeout');
+        }
+        argv.splice(index, 2);
+      } else {
+        invalidArgs.add('--request-timeout');
+        argv.splice(index, 1);
+      }
+    }
+
+    // process proxy host
+    index = argv.indexOf('--proxy-host');
+    if (index !== -1) {
+      if (CommandLineManager.validArgValue(argv[index + 1])) {
+        RdGlobalConfig.proxy = RdGlobalConfig.proxy || {};
+        RdGlobalConfig.proxy.host = argv[index + 1];
+        argv.splice(index, 2);
+      } else {
+        invalidArgs.add('--proxy-host');
+        argv.splice(index, 1);
+      }
+    }
+
+    // process proxy port
+    index = argv.indexOf('--proxy-port');
+    if (index !== -1) {
+      if (CommandLineManager.validArgValue(argv[index + 1])) {
+        if (RdGlobalConfig.proxy && RdGlobalConfig.proxy.host) {
+          RdGlobalConfig.proxy.port = argv[index + 1];
+        } else {
+          if (!invalidArgs.has('--proxy-host')) missingArgs.add('--proxy-host');
+        }
+        argv.splice(index, 2);
+      } else {
+        invalidArgs.add('--proxy-port');
+        argv.splice(index, 1);
+      }
+    }
+    
+    // if proxy port value in invalid or doesn't exist and host exists, set the default value
+    if (RdGlobalConfig.proxy && RdGlobalConfig.proxy.host && (invalidArgs.has('--proxy-port') || !RdGlobalConfig.proxy.port)) {
```

Comment:
Any specific reason ignoring incorrect port value. 
From user point of view, he is passing 666666 as port and its not being used.

Reply:
Discussed f2f. We log it with the default value that was assigned when the tool boots up.

Reply:
Added the proxy-port check. That will take care of this scenario.

Reply:
Looks good

Reply:
Resolving.